datalad/support/tests/test_locking.py

```diff
@@ -98,3 +106,74 @@ def test_lock_if_check_fails(tempfile):
     p.start()
     ok_(q.get())
     p.join()
+
+
+@with_tempfile
+def test_try_lock_informatively(tempfile):
+    lock = InterProcessLock(tempfile + '.lck')
+    lock_path = ensure_unicode(lock.path)  # can be bytes, complicates string formattingetc
+    t0 = time()
+    with try_lock_informatively(lock, purpose="happy life") as acquired:
+        assert_true(lock.acquired)
+        assert_true(acquired)
+        assert_greater(2, time() - t0)  # should not take any notable time, we cannot be blocking
+
+        """
+        # InterProcessLock is not re-entrant so nesting should not be used, will result
+        # in exception on release
+        with try_lock_informatively(lock, timeouts=[dt, dt*2], proceed_unlocked=True) as acquired:
+            assert_true(lock.acquired)  # due to outer cm
+            assert_true(acquired)       # lock is reentrant apparently
+        """
+        # Let's try in a completely different subprocess
+        runner = WitlessRunner(env=dict(os.environ, DATALAD_LOG_LEVEL='info', DATALAD_LOG_TARGET='stderr'))
+
+        script1 = Path(tempfile + "-script1.py")
+        script1_fmt = f"""
+from fasteners import InterProcessLock
+from time import time
+
+from datalad.support.locking import try_lock_informatively
+
+lock = InterProcessLock({lock_path!r})
+
+with try_lock_informatively(lock, timeouts=[0.05, 0.15], proceed_unlocked={{proceed_unlocked}}) as acquired:
+    print("Lock acquired=%s" % acquired)
+"""
+        script1.write_text(script1_fmt.format(proceed_unlocked=True))
+        t0 = time()
+        res = runner.run([sys.executable, str(script1)], protocol=StdOutErrCapture)
+        assert_in('Lock acquired=False', res['stdout'])
+        assert_in(f'Failed to acquire lock at {lock_path} in 0.05', res['stderr'])
+        assert_in(f'Failed to acquire lock at {lock_path} in 0.15', res['stderr'])
+        assert_in('proceed without locking', res['stderr'])
+        assert_greater(time() - t0, 0.19999)  # should wait for at least 0.2
+        try:
+            # PID does not correspond
+            assert_in('Check following process: PID=', res['stderr'])
+            assert_in(f'CWD={os.getcwd()} CMDLINE=', res['stderr'])
+        except AssertionError:
+            # we must have had the other one then
+            assert_in('failed to determine one', res['stderr'])
```

Comment:
note: no coverage here is odd since it should be happening on OSX... aren't we pushing coverage from those???

Reply:
hm, looking at `/mnt/btrfs/datasets/datalad/ci/logs/2021/08/12/pr/5884/9827dc3/github-Test on macOS-1723-success/test (brew)/9_Upload coverage report to Codecov.txt` -- coverage is successfully uploaded, and points to https://codecov.io/github/datalad/datalad/commit/9827dc38edf72106d429ee78dfa8d632a92bed43 but that one is not "run specific".  I wonder if combining from multiple submissions works correctly, or just the last one to arrive (travis? ;)) wins it all?